connectors/agent/config.py

```diff
@@ -74,7 +79,12 @@ def try_update(self, source):
             }
 
             if source.fields.get("api_key"):
-                es_creds["api_key"] = source["api_key"]
+                api_key = source["api_key"]
+                # if beats_logstash_format we need to base64 the key
+                if ":" in api_key:
+                    api_key = base64.b64encode(api_key.encode()).decode()
```

Comment:
I thought that @jedrazb had discovered that encode/decode isn't necessary - you actually want to just grab the bit after the colon?

Reply:
Checked with Jedr:

encode/decode are for string operations here.
`base64.b64encode` wants a byte array, so we `encode` and returns a byte array, so we `decode` to get a string out of it

Reply:
> I thought that @jedrazb had discovered that encode/decode isn't necessary - you actually want to just grab the bit after the colon?

The colon indicates that the API key we’re receiving is in an unencoded `beats_logstash_format`, whereas the ES client expects it to be base64-encoded. Therefore, we need to handle the encoding ourselves to authenticate with the ES client.

IMO checking for the presence of a `:` is an "acceptable" way to detect the API key format since `:` isn’t a character used in base64-encoded strings.